File: c22_1_binomial_coefficients.py
import unittest
from math import comb

def scenario1(n,k):
    return comb(n,k)

def scenario2(n,k):
    return comb(k+n-1,k)

def scenario3(n,k):
    return comb(n-k+1,n-2*k+1)

class Test(unittest.TestCase):
    def test(self):
        self.assertEqual(scenario1(5,2),10)
        self.assertEqual(scenario2(5,2),15)
        self.assertEqual(scenario3(5,2),6)

if __name__=='__main__':
    unittest.main()

File: c++/c22_1_binomial_coefficients.cpp
#include <bits/stdc++.h>
#include <gtest/gtest.h>
using namespace std;
using ll = long long;
#include "util.h"

namespace {
    ll scenario1(ll n, ll k) {
        return comb(n, k);
    }

    ll scenario2(ll n, ll k) {
        return comb(k + n - 1, k);
    }

    ll scenario3(ll n, ll k) {
        return comb(n - k + 1, n - 2 * k + 1);
    }
}

TEST(C221BinomialCoefficients, scenario1) {
    EXPECT_EQ(scenario1(5, 2), 10);
}

TEST(C221BinomialCoefficients, scenario2) {
    EXPECT_EQ(scenario2(5, 2), 15);
}

TEST(C221BinomialCoefficients, scenario3) {
    EXPECT_EQ(scenario3(5, 2), 6);
}
